### PR TITLE
Connect: Prepare keyboard shortcuts configuration

### PR DESCRIPTION
**Pull request by gzdunek**
1/2
Part of https://github.com/gravitational/teleport/issues/21914

Initially I wanted to create a single PR for all the changes, but I realized that it will be hard to review. The PR that actually enables shortcuts configuration is [here](https://github.com/gravitational/teleport/pull/22194).

In this PR I focused on renaming a few things:
- For keyboard shortcuts, from now we will be using only "accelerator" (e.g. `Command+1`) and "action" (e.g. `tab1`) words. Previously, we used words like: "key", "type", "shortcut" which were really confusing (even for me who wrote this code 🙃). I borrowed "accelerator" [from Electron](https://www.electronjs.org/docs/latest/api/global-shortcut#methods).
- I renamed some config keys, like `keymap.tabNew`, to `keymap.newTab` to make them all more consistent.
- I renamed some config keys, like `keymap.toggleClusters` to `keymap.openClusters`. "Toggle" sounds like a more serious and permanent action, while it only opens (and closes) a popup.
- I renamed keyboard shortcuts actions, like `tab-1` to `tab1` to match config keys (`keymap.tab1`).

I also changed the separator for the accelerators. With "-" we were not able to use this symbol as a valid key code.

Best to review commit by commit.

### CODE REVIEW

File: web/packages/teleterm/src/ui/services/keyboardShortcuts/useKeyboardShortcutFormatters.ts [hunk 1]
```diff
@@ -41,18 +41,18 @@ export function useKeyboardShortcutFormatters(): KeyboardShortcutFormatters {
   const keyboardShortcuts = keyboardShortcutsService.getShortcutsConfig();
 
   return {
-    getLabelWithShortcut(label, shortcutKey, options) {
+    getLabelWithShortcut(label, shortcutAction, options) {
```
Comment (ravicious) [inline web/packages/teleterm/src/ui/services/keyboardShortcuts/useKeyboardShortcutFormatters.ts:44]: Should those functions be renamed to something like `getLabelWithAccelerator`, `getAccelerator`? I like the action & accelerator distinction, when we say just "shortcut" I feel like we loose the clarity that this distinction brings us.

File: web/packages/teleterm/src/ui/services/keyboardShortcuts/useKeyboardShortcutFormatters.ts [hunk 2]
```diff
@@ -61,19 +61,19 @@ export function useKeyboardShortcutFormatters(): KeyboardShortcutFormatters {
 
 function formatKeyboardShortcut(options: {
```
Comment (ravicious) [inline web/packages/teleterm/src/ui/services/keyboardShortcuts/useKeyboardShortcutFormatters.ts:62]: Same question as above.

**Review by ravicious — APPROVED** [review]
Good refactor, I like the distinction between an accelerator and an action a lot.

At first I thought that the name "accelerator" is just a pun (because electron accelerators exist) but it looks like [it's used outside of Electron as well](https://learn.microsoft.com/en-us/windows/apps/design/input/keyboard-accelerators).

File: web/packages/teleterm/src/ui/services/keyboardShortcuts/useKeyboardShortcutFormatters.ts (hunk 1 above)
Comment (gzdunek) [reply web/packages/teleterm/src/ui/services/keyboardShortcuts/useKeyboardShortcutFormatters.ts:44]: Done! I'm wondering if we should rename the top level `useKeyboardShortcutFormatters`. It feels to me like it is a good, more general name 🤔

File: web/packages/teleterm/src/ui/services/keyboardShortcuts/useKeyboardShortcutFormatters.ts (hunk 2 above)
Comment (gzdunek) [reply web/packages/teleterm/src/ui/services/keyboardShortcuts/useKeyboardShortcutFormatters.ts]: Done

File: web/packages/teleterm/src/ui/services/keyboardShortcuts/keyboardShortcutsService.ts [hunk 3]
```diff
@@ -117,15 +118,14 @@ export class KeyboardShortcutsService {
...
-   */
-  private recalculateKeysToShortcuts(
-    toInvert: Partial<KeyboardShortcutsConfig>
-  ): void {
-    this.keysToShortcuts.clear();
-    Object.entries(toInvert).forEach(([shortcutType, key]) => {
-      this.keysToShortcuts.set(key, shortcutType as KeyboardShortcutType);
+  /** Inverts shortcuts-keys pairs to allow accessing shortcut by an accelerator. */
+  private mapAcceleratorsToActions(): void {
+    this.acceleratorsToActions.clear();
```
[the remark this answers is not in the document]
Comment (gzdunek) [reply web/packages/teleterm/src/ui/services/keyboardShortcuts/keyboardShortcutsService.ts]: >Why does it need to call clear in the first place? 🤔 

I think it was prepared to recalculate the shortcuts multiple times (to update them while the app is working), but we do it only once :)

>I'd be for rewriting it to be free of side effects.

Done.

**Review by avatus — APPROVED** [review]

File: web/packages/teleterm/src/ui/services/keyboardShortcuts/useKeyboardShortcutFormatters.ts (hunk 1 above)
Comment (ravicious) [reply web/packages/teleterm/src/ui/services/keyboardShortcuts/useKeyboardShortcutFormatters.ts]: > I'm wondering if we should rename the top level `useKeyboardShortcutFormatters`. It feels to me like it is a good, more general name 🤔

I feel the same way.

**Pull request merged**